storage/store/inmemory/unsynchronized/events.go

```diff
@@ -0,0 +1,129 @@
+package unsynchronized
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
+
+type Events struct {
+	//TODO: we don't need a mutex here as we have a guarantee by design
+	// that we write data only once and it happens before the future reads.
+	// We decided to leave a mutex for some time during active development.
+	// It'll be removed in the future.
+	lock            sync.RWMutex
+	blockIdToEvents map[flow.Identifier][]flow.Event
+}
+
+var _ storage.Events = (*Events)(nil)
+
+func NewEvents() *Events {
+	return &Events{
+		blockIdToEvents: make(map[flow.Identifier][]flow.Event),
+	}
+}
+
+// ByBlockID returns the events for the given block ID.
+//
+// Expected errors during normal operation:
+//   - `storage.ErrNotFound` if no events were found at given block.
+func (e *Events) ByBlockID(blockID flow.Identifier) ([]flow.Event, error) {
+	e.lock.RLock()
+	defer e.lock.RUnlock()
+
+	val, ok := e.blockIdToEvents[blockID]
+	if !ok {
+		return nil, storage.ErrNotFound
+	}
+
+	return val, nil
+}
+
+// ByBlockIDTransactionID returns the events for the given block ID and transaction ID.
+//
+// Expected errors during normal operation:
+//   - `storage.ErrNotFound` if no events were found at given block and transaction.
+func (e *Events) ByBlockIDTransactionID(blockID flow.Identifier, txID flow.Identifier) ([]flow.Event, error) {
+	events, err := e.ByBlockID(blockID)
+	if err != nil {
+		return nil, storage.ErrNotFound
```

Comment:
I think this should just pass the error through to avoid issues if `ByBlockID` is changed in the future
```suggestion
		return nil, err
```